pkg/identityserver/bunstore/contact_info_store.go

```diff
@@ -534,6 +561,31 @@ func (s *contactInfoStore) GetValidation(
 	return validationToPB(model), nil
 }
 
+func (s *contactInfoStore) ListRefreshableValidations(
+	ctx context.Context, pb *ttnpb.EntityIdentifiers, refreshInterval time.Duration,
+) ([]*ttnpb.ContactInfoValidation, error) {
+	ctx, span := tracer.StartFromContext(ctx, "ListRefreshableValidations", trace.WithAttributes(
+		attribute.String("entity_type", pb.EntityType()),
+		attribute.String("entity_id", pb.IDString()),
+		attribute.String("refresh_interval", refreshInterval.String()),
+	))
+	defer span.End()
+
+	entityType, entityUUID, err := s.getEntity(ctx, pb)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.listValidationsBy(ctx, func(q *bun.SelectQuery) *bun.SelectQuery {
+		return q.
+			Where("?TableAlias.entity_id = ?", entityUUID).
+			Where("?TableAlias.entity_type = ?", entityType).
+			Where("?TableAlias.used IS NULL OR ?TableAlias.used = false").
+			Where("?TableAlias.expires_at IS NULL OR ?TableAlias.expires_at > NOW()").
+			Where("?TableAlias.updated_at <= ?", now().Add(-refreshInterval))
```

Comment:
I would do the arithmetic using `NOW()` in SQL, not by providing the timestamp here.